### DISCUSSION TITLE
Reverse Quadratic Optimization with Linear constraints

### DISCUSSION THREAD

**Discussion by 0xF0F0F0F** [yalmip/YALMIP#1265] [Optimization modelling] [answered]
Considering following quadratic optimization problem with linear equality constraints
min_w    1/2 w'**Q**w - **r**'w
s.t.             **A**w <= **b**
By given vector **r** and positive definite **Q**, it is easy to find the solution using Yalmip.

But, I was wondering, by given the optimal w∗, is it possible to derive the vector r using Yalmip (i.e., for unconstrainted QP problem r = Qw*) ?

Many thanks!

**Comment by johanlofberg** [comment]
It is called inverse programming. You basically derive the dual (i.e. kkt conditions etc), and the optimize over r and the duals, with the given solution plugged in.

**Reply by 0xF0F0F0F** [reply, marked as answer]
Thank you very much for your replying! 👍  Yesterday I tried with only one equality constraint: sum(w) = 1, but it seems to have ranking problem, I posted this question in stackexchange:
[https://math.stackexchange.com/questions/4683578/inverse-quadratic-optimization-with-linear-equality-constraints/4683635?noredirect=1#comment9905675_4683635](url)

**Comment by johanlofberg** [comment]
Something like (this sets up a problem where r isn't necessarily unique in the inverse problem)

```
n = 3;
m = 9;
Q = randn(n);Q = Q*Q';
r = randn(n,1);
rstart = r;
A = randn(m,n);
b = rand(m,1)*n;

w = sdpvar(n,1);
optimize(A*w <= b,.5*w'*Q*w-r'*w)
wsol = value(w);

r = sdpvar(n,1);
z = sdpvar(m,1);
optimize([z>=0,z'*(b-A*wsol)==0, Q*wsol-r-A'*z==0],-r'*wsol)
% Not necessarily the same
[rstart value(r)]

% Verify solution is same
optimize(A*w <= b,.5*w'*Q*w-value(r)'*w)
[wsol value(w)]

```

**Reply by 0xF0F0F0F** [reply]
Many thanks johan it helps a lot!

**Reply by johanlofberg** [reply]
The model you discuss on stackexchange is trivially non-unique in r (thus making the matrix singular, which seem to be your concern there). I will post an answer there
